typed-racket-test/unit-tests/typecheck-tests.rkt

```diff
@@ -425,6 +425,29 @@
         (tc-e (expt -0.0 -1.0) -NonNegFlonum)
         (tc-e (expt 0 (flabs (cos (real->double-flonum 2))))
               -Real)
+        (tc-err ;; Github issue #115
+          ((lambda ([x : Nonpositive-Flonum]) x)
```

Comment:
At first I tried using `ann`, but then the tests didn't work. Is there a better way to give an incorrect type that I want to assert fails?

For example if I replace `lambda` with `ann` in this test I get:

```
Typechecker tests > tc-expr tests > 429 (FAIL (ann (let ((x : Flonum 5.0)) (if (fl>= x (ann -4.0 Nonpositive-Flonum)) x -3.0)) Nonpositive-Flonum))
429 (FAIL (ann (let ((x : Flonum 5.0)) (if (fl>= x (ann -4.0 Nonpositive-Flonum)) x -3.0)) Nonpositive-Flonum))
FAILURE
location:   <pkgs>/typed-racket-test/unit-tests/typecheck-tests.rkt:428:8
actual:     (tc-results
 (list
  (tc-result
   (U Float-Nan Float-Positive-Zero Float-Negative-Zero Negative-Float-No-NaN)
   (Top | Bot)
   -))
 #f)
expected:   (tc-results (list (tc-result (U) (Bot | Bot) -)) #f)
tc-expr did not return the expected value. The types don't match.
```